### DISCUSSION TITLE
[Solved] Module cannot be loaded

### DISCUSSION THREAD

**Discussion by shugal93** [HotCakeX/Harden-Windows-Security#305] [Q&A] [answered]
I ran a compliance check today and everything worked. But then suddenly the module couldn´t be loaded. I've restarted my PC, but the problem persists. The 'Import-Module Harden-Windows-Security-Module' doesn't work either. Any suggestions?
<img width="667" alt="PowerShell" src="https://github.com/user-attachments/assets/17cc93d2-37ff-4fca-a616-18b957f0d31e">

**Comment by HotCakeX** [comment]
What do you see when you run this command?
```powershell
Import-Module -Name 'Harden-Windows-Security-Module'
```

**Reply by shugal93** [reply]
This: 
<img width="664" alt="PowerShell import" src="https://github.com/user-attachments/assets/841384d5-a9fb-4227-a1d5-77c8f78ac1bf">

**Reply by HotCakeX** [reply]
@shugal93 
I'm trying to reproduce that error but i can't. I tried installing Spanish language pack to change the entire Windows language to another language that is not English because i thought maybe it could be related but still nothing.

Tested on different systems too. I already can think of ways to improve the code but without knowing how to trigger that error i can't be sure it is fixed.

So can you please show me the output of the following commands:

```powershell
Get-History
```

and 

```powershell
cat "$PSHome/pwsh.runtimeconfig.json"
```

I also noticed your starting directory is System32, which is the default directory of the old Windows PowerShell so i'm assuing you have a config in your PS profile that does that.

and if there is anything else you configured on your environment that might cause that error it'd be helpful to share, thanks

**Reply by shugal93** [reply]
I installed PS with Winget and now it's not System32 anymore. This is the outputs of the commands:
<img width="476" alt="PowerShell history" src="https://github.com/user-attachments/assets/71736979-a03a-4112-831e-c36ea4533a40">
I don´t think I've done anything configurations since hardening my system. It worked fine the first time I applied the security module.

**Reply by HotCakeX** [reply]
Thank you, can you also open a new PowerShell tab, reproduce that error and then use this cmdlet?
```powershell
Get-Error
```
I need the output of that bcz im hoping it will include more info to pinpoint the issue, thanks

**Reply by shugal93** [reply]
Sure! 
<img width="662" alt="PS Error II" src="https://github.com/user-attachments/assets/8e68ad16-c9e1-4e6c-817d-dfcf05fd8dc3">

**Reply by Jaykul** [reply]
I think he needed the `Get-Error` output after `Import-Module` because _that_ isn't showing the error that happened inside the module. You might be able to `Get-Error 3` to see other errors, but I think PowerShell is going to hide the import error unless you explicitly import.

**Reply by HotCakeX** [reply]
Can u paste the text please?

use 
```powershell
import-Module -Name 'Harden-Windows-Security-Module'
```

and then when the error is shown, use 
```powershell
Get-Error
```

and copy everything displayed and paste here.

Also can u show me the output of this command too? I think im finding out the root cause and this info will confirm it.

```powershell
Get-Culture
```

**Comment by shugal93** [comment]
Well, I hope this is what you need:
<img width="826" alt="PS Import I" src="https://github.com/user-attachments/assets/57e8644f-f4a6-4880-9516-f6dab9e82713">
<img width="674" alt="PS Import II" src="https://github.com/user-attachments/assets/abe1a770-9761-4fb2-b46f-b3933b7a4a57">
<img width="290" alt="PS Culture" src="https://github.com/user-attachments/assets/88f26a05-dea7-41ca-a873-eb75ab78bf60">

**Comment by HotCakeX** [comment, marked as answer]
@shugal93 thank you, i've fixed the problem and will release an update soon. If you're interested, the root cause is that in your language pack which is Swedish, the decimal separator is `,` and In the code i used `.` and that is causing the issue.

If you want to temporarily make it work, you can switch to another language pack such as English that uses `.` as decimal separator until the update is released (hopefully by the end of tomorrow)

Thanks again 🙏

**Reply by shugal93** [reply]
Well, that's wonderful news! I'm glad my problem helped you solve a problem (win win). Thanks!

**Reply by HotCakeX** [reply]
For sure! 🙏🙏

**Comment by HotCakeX** [comment]
Hi, I fixed the issue in version 0.5.0
https://github.com/HotCakeX/Harden-Windows-Security/releases/tag/Hardening-Module-v.0.5.0

I was planning on shipping it with the rest of the changes i've been working on but i would have to wait much longer to do the validations on the new code so instead released the v0.5.0 as a small update that only takes care of this issue.

Please use the following command to install the new version
```powershell
Install-Module -Name 'Harden-Windows-Security-Module' -Force
```
